text/002-css-vars.md

```diff
@@ -0,0 +1,45 @@
+- Start Date: 10-10-2023
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+
+Describes how we use CSS custom properties or CSS variables in the builder.
+
+# Motivation
+
+1. Avoid namespace collisions
+2. Avoid copy-pasting CSS variable names
+3. Provide a way to define a typed interface for CSS variables
+
+CSS variables are a subject to inheritance, hence in a large app you can get conflicts. In addition to that not all values should be allowed and we should be able to trace what code is setting a value.
+By defining a convention and using a TS interface we can solve both problems.
+
+# Detailed design
+
+1. Avoiding namespace collision is solved by using a naming convention: `--ws-{component}-{var}` where `component` is the component name that is going to render the variable and `var` is the variable namespace.
```

Comment:
one question here is if it should be a component name or a module name, because sometimes you will need the same var in multiple components within the same module